### PR TITLE
chore(storybook): support static color settings in testing grid

### PR DESCRIPTION
**Pull request by castastrophe**
## Description

This update adds support for static colors in the testing grids saving us from having to run static white/static black snapshots separately. I accomplished this by updating which containers are returned by the fetchContainers function based on if the document is in a testing state or not.

Updated tests:

- ActionButton: Remove separate static black & static white tests; add both to the testing grid.
- ActionGroup: Add static black & static white to the test grid.
- Button: Remove separate static black & static white tests; add both to the testing grid.
- ClearButton: Remove OverBackground test; add static white to testing grid.
- CloseButton: Add static black & static white to testing grid.
- Divider: Remove static white and static black from separate snapshots; add to testing grid.
- Link: Remove static white and static black and static white + quiet and static black + quiet from separate snapshots; add to testing grid.
- ProgressBar: Remove static white from test; add to testing grid.
- Meter: Remove static white from test; add to testing grid (remove the skipBorders flag).
- ProgressCircle: Remove static white from test; add to testing grid.

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [ ] On the [action button story](https://64762974a45b8bc5ca1705a2-kbpzmrxpgp.chromatic.com/?path=/story/components-action-button--default&args=staticColor:white), expect to see a dark teal background with a grouping of white outlined buttons
- [ ] In the [action button testing grid](https://64762974a45b8bc5ca1705a2-kbpzmrxpgp.chromatic.com/?path=/story/components-action-button--default&globals=testingPreview:!true), expect to see a light teal and dark teal box containing the static white and static black variant tests: 

Screenshot 2024-09-27 at 6 31 41 PM

- [x] Review the [changeset](https://www.chromatic.com/review?appId=64762974a45b8bc5ca1705a2&number=22&type=unlinked&view=changes) for this PR to validate other additions to the snapshots

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [x] VRTs have been run and looked at.
- [x] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] I have tested these changes in Windows High Contrast mode.
- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.
- [x] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

**Review by rise-erpelding — COMMENTED** [review]
This is awesome! Static color tests look great, a couple of questions with those updates:

We shouldn't be expecting any differentiation between static black and static white for forced colors, should we?
image

It looks like progress bar/meter aren't showing the fill for forced colors, is that something we should be following up on?
image

Taking a look at the resize observer, this appears to work great! However, the Medium/Large storybook control doesn't appear to be doing much anymore, does it make sense to take it out as part of this work?

**Comment by castastrophe** [comment]
> We shouldn't be expecting any differentiation between static black and static white for forced colors, should we? (My guess is no, but just want to be sure.) 

Yes, you are correct. No expected differentiation for static black/static white in forced colors mode is expected.

> It looks like progress bar/meter aren't showing the fill for forced colors, is that something we should be following up on? 

Yes!  Let's open a Jira ticket to cover that.

> Taking a look at the resize observer, this appears to work great! However, the Medium/Large storybook control doesn't appear to be doing much anymore, does it make sense to take it out as part of this work?

I think there's value in the user being able to force Medium/Large mode on-demand but the 2 could definitely use a tighter coupling.  I'll add a Jira ticket to update the global arg if the screen size changes the scale.

**Review by rise-erpelding — APPROVED** [review]
My main concern was that the scale control is broken here, but if we plan to fix that elsewhere, that will work! Static colors look great in the testing grid.

**Comment by castastrophe** [comment]
@rise-erpelding I decided to pull out the resize observer toggling the scale and let this PR focus on the static color support.

File: components/button/stories/button.test.js [hunk 1]
```diff
@@ -47,6 +40,14 @@ export const ButtonGroups = Variants({
 			variant,
 			treatment: "outline",
 		})),
+		{
+			testHeading: "Static black",
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply components/button/stories/button.test.js:44]: Oh good point! I can add an ignore to the isPending state for "Static black" and that should leave that off until we have styles for it.

File: .storybook/decorators/context.js [hunk 2]
```diff
@@ -36,36 +37,53 @@ export const withContextWrapper = makeDecorator({
...
 				// Check if the container has a static color element
-				const hasStaticElement = container.matches(`:has(.${rootClass}--staticWhite, .${rootClass}--staticBlack, .${rootClass}--overBackground)`);
+				let hasStaticElement = container.matches(`:has([data-html-preview])`) ? container.matches(`:has([data-html-preview] .${rootClass}--staticWhite, [data-html-preview] .${rootClass}--staticBlack, [data-html-preview] .${rootClass}--overBackground)`) : container.matches(`:has(.${rootClass}--staticWhite, .${rootClass}--staticBlack, .${rootClass}--overBackground)`);
+
 				let staticKey = staticColor;
 				if (!staticKey && hasStaticElement) {
 					staticKey = (
-						container.querySelector(`.${rootClass}--staticWhite`) && "white" ||
-						container.querySelector(`.${rootClass}--staticBlack, .${rootClass}--overBackground`) && "black"
+						container.querySelector(`.${rootClass}--staticWhite, .${rootClass}--overBackground`) && "white" ||
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply .storybook/decorators/context.js]: This actually inspired me to add the staticWhite class to the clear button template even if the styles aren't attached to that, we can still reduce the complexity of the static color selectors by pretending overBackground doesn't exist. :)

File: components/actionbutton/stories/actionbutton.test.js [hunk 3]
```diff
@@ -4,24 +4,27 @@ import { Template } from "./template.js";
...
+				iconName: undefined,
+			}, context)}
+			${Template({
+				...args,
+				hideLabel: true,
+			}, context)}
+			${Template({
+				...args,
+				hasPopup: "true",
+                hideLabel: true,
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply components/actionbutton/stories/actionbutton.test.js]: Can do!

File: components/clearbutton/stories/template.js [hunk 4]
```diff
@@ -21,10 +21,14 @@ export const Template = ({
 		type="reset"
 		class=${classMap({
 			[rootClass]: true,
-			[`${rootClass}--size${size?.toUpperCase()}`]:
-				typeof size !== "undefined",
+			[`${rootClass}--size${size?.toUpperCase()}`]: typeof size !== "undefined",
 			[`${rootClass}--quiet`]: isQuiet,
 			[`${rootClass}--overBackground`]: staticColor === "white",
+			/**
+			 * There aren't styles for `--staticWhite` in clear button (yet)
```
Comment (marissahuysentruyt) [inline components/clearbutton/stories/template.js:28]: *yet* - love it!

**Review by marissahuysentruyt — APPROVED** [review]
Looks wonderful!

spongebob-thumbsup

**Pull request merged**